### PR TITLE
Endpoint for GET Users

### LINKED ISSUE #31: Backend: Endpoint to retrieve users - GET api/users
## User story

**Who:** **As a** developer
**What:** **I want to** be able to get users with specific criteria
**Why:** **so that we can** show those users

## Implementation details
- Create the `GET` endpoint for `/api/users`

**Think about ways to filter the questions that make sense. These should be added as query parameters. Ask if this does not make sense**

Remember to split the story into smaller PR's! Maybe you can do a PR with just the words, filter fx. You can also choose to do it in separate commits. just be very aware of the PR's not becoming too big!

Check the README file here https://github.com/HackYourFuture-CPH/guide-it#-working-with-api-documentation---swagger and this article should also help you with that https://blog.logrocket.com/documenting-your-express-api-with-swagger/

### PR DESCRIPTION
**Pull request by Poorooshka**
# Description

We want to be able to get users with specific criteria so that we can show those users. I created a GET request for all the users to retrieve a user by id.

Fixes #31 

# How to test?

Once you have run the migration and seeding to your local database, open this branch and run npm run dev. Go to localhost:3000/api/documentation. You will see two GET requests for users. You can run the tests from there, to retrieve all the users and a user by id.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Comment by ditlevHYF** [comment]
Hello Nashmil, please rename the PR with a more descriptive name and a reference to a userstory.

**Comment by Poorooshka** [comment]
> Hello Nashmil, please rename the PR with a more descriptive name and a reference to a userstory.

Hello ditlev. I made changes. Hope it works fine now :)

**Comment by Poorooshka** [comment]
I did resolve the conflict and now it's ready to be merged.

**Review by TITHI1244 — APPROVED** [review]
We have checked this GET endpoint code for users table in group 4 and it worked fine.

File: src/server/api/controllers/users.controller.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  try {
+    const users = await knex('users')
+      .select('users.id as id', 'full_name', 'firebase_token')
+      .where({ id });
+    if (users.length === 0) {
+      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    }
+    return users;
```
Comment (bytewiz) [inline src/server/api/controllers/users.controller.js:24]: if you only return one user it is should be named `user` and not `users` as that might be confusing :)

Comment (bytewiz) [reply src/server/api/controllers/users.controller.js:24]: and I would expect to receive an object of that user and not an array

**Review by bytewiz — APPROVED** [review]
that also works 👌

**Comment by wwang107** [comment]
Looks good!

**Pull request merged**